src/search.js

```diff
@@ -0,0 +1,231 @@
+"use strict";
+
+var path = require("path");
+var fs = require("fs");
+var moment = require("moment");
+var Promise = require("bluebird");
+var Sqlite = require("sqlite3");
+var fsextra = require("fs-extra");
+var es = require("event-stream");
+var Helper = require("./helper");
+
+try {
+	fsextra.ensureDirSync(Helper.getLogsPath());
+} catch (e) {
+	log.error("Unable to create log directory", e);
+	return;
+}
+
+var format = Helper.config.logs.format || "YYYY-MM-DD HH:mm:ss";
+
+var schema = [
+	"CREATE TABLE IF NOT EXISTS logs (user TEXT, network TEXT, chan TEXT, time INTEGER, nick TEXT, text TEXT)",
+	"CREATE VIRTUAL TABLE IF NOT EXISTS texts USING fts4(content=\"logs\", nick, text)",
+	"CREATE TRIGGER IF NOT EXISTS logs_bu BEFORE UPDATE ON logs BEGIN DELETE FROM texts WHERE docid=old.rowid; END",
+	"CREATE TRIGGER IF NOT EXISTS logs_bu BEFORE UPDATE ON logs BEGIN DELETE FROM texts WHERE docid=old.rowid; END",
+	"CREATE TRIGGER IF NOT EXISTS logs_bd BEFORE DELETE ON logs BEGIN DELETE FROM texts WHERE docid=old.rowid; END",
+	"CREATE TRIGGER IF NOT EXISTS logs_ai AFTER INSERT ON logs BEGIN INSERT INTO texts(docid, nick, text) VALUES(new.rowid, new.nick, new.text); END"
+];
+
+var db = new Sqlite.Database(path.join(Helper.getLogsPath(), "search.db"));
+db.serialize(() => schema.forEach((line) => db.run(line)));
+
+var indexAll = () => {
```

Comment:
Since these index functions are only used from CLI, could you move them there?

Reply:
I rather prefer keep code that have access to the "db" object (and do queries) in the same file.